### ISSUE TITLE
[Issue]: Is there an existing Entity Resolution step?

### ISSUE THREAD

**Issue by mayukhb02** [microsoft/graphrag#847]
### Is there an existing issue for this?

- [X] I have searched the existing issues
- [X] I have checked [#657](https://github.com/microsoft/graphrag/issues/657) to validate if my issue is covered by community support

### Describe the issue

Does GraphRAG have a default implementation of Entity Resolution currently? I read [here](https://microsoft.github.io/graphrag/posts/index/1-default_dataflow/#:~:text=Entity%20Resolution%20(Not%20Enabled%20by%20Default)) that it does. I tried activating it by setting `GRAPHRAG_ENTITY_RESOLUTION_ENABLED=True` but it didn't seem to make any difference. I am getting the same number of entities in `create_final_entities.parquet`. I also don't see any mention of `resolve_entities` step in the list of steps shown on terminal.

If it is implemented, is there something extra that needs to be modified in the configs? Or is there a way to verify that it has run through the logs?

Relevant issue: https://github.com/microsoft/graphrag/issues/401

### GraphRAG Config Used

```yaml
# Paste your config here

```

### Additional Information

- GraphRAG Version:
- Operating System:
- Python Version:
- Related Issues:

**Comment by natoverse** [comment]
There is not. We had one during development, but weren't yet satisfied with the consistency so it was removed. There may be leftover cruft that wasn't cleaned out. We may work on this again in the future, but don't have a planned date.

**Comment by mayukhb02** [comment]
Thanks for letting me know

**Comment by saaspeter** [comment]
The document states: It is done via LLM, It will provide the LLM with a series of entities and ask it to determine which ones should be merged. so what's the solution? for one entity loop all other entities, then ask LLM whether they are same entity? (Does means if we have N entities, then need call n*(n-1)*(n-2)...*2 times LLM api?)

Is there good and more faster and effective solution for the entity Resolution step?

**Comment by PetrosTragoudaras** [comment]
Hello all,

Do we have an update on this?
Anything done or on the works?
We would love to have that feature ready and use it in our project.

Thank you in advance!

**Comment by jules-gd-dev** [comment]
I ran into the same issue with LangChain extraction creating fragmented nodes. I built AutoGraft, an open-source Entity Resolution middleware that intercepts entities before they hit the graph and resolves duplicates locally using RapidFuzz + vector similarity. On a 500-doc benchmark it caught 2869/2869 deterministic duplicates at zero tokens, only invoking the LLM for 7 ambiguous cases. It plugs directly into LangChain/LlamaIndex pipelines: https://github.com/jules-gd-dev/autograft-lib
